File: Java/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        
        char[] primitiveChar = s.toCharArray();
        for (char c: primitiveChar) {
            if (c == '(') {
                stack.push(c);
            } else {
                if (stack.isEmpty() || stack.pop() == c) return false;
            }
        }

        return stack.isEmpty();
    }
}

File: Swift/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.swift
import Foundation

struct Stack<T> {
    private var stack: [T] = []
    
    var isEmpty: Bool {
        return stack.isEmpty
    }
    
    var count: Int {
        return stack.count
    }
    
    mutating func push(_ element: T) {
        stack.append(element)
    }    
    
    mutating func pop() -> T? {
        return isEmpty ? nil : stack.popLast()
    }
}

func solution(_ s:String) -> Bool
{
    var stack = Stack<Character>()

    for char in s {
        if char == "(" {
            stack.push(char)
        } else {
            guard stack.pop() != nil else { return false }
        }
    }

    return stack.isEmpty
}